File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/playback/file/initialization/IPlaybackInitialization.java
package com.lasthopesoftware.bluewater.client.playback.file.initialization;

import android.net.Uri;

import java.io.IOException;

/**
 * Created by david on 9/24/16.
 */

public interface IPlaybackInitialization<TMediaPlayer> {
	TMediaPlayer initializeMediaPlayer(Uri fileUri) throws IOException;
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/playback/file/initialization/IPlaybackInitialization.kt
package com.lasthopesoftware.bluewater.client.playback.file.initialization

import android.net.Uri
import java.io.IOException

interface IPlaybackInitialization<TMediaPlayer> {
    @Throws(IOException::class)
    fun initializeMediaPlayer(fileUri: Uri?): TMediaPlayer
}
